mesonbuild/utils/universal.py

```diff
@@ -1587,13 +1587,12 @@ def Popen_safe_logged(args: T.List[str], msg: str = 'Called', **kwargs: T.Any) -
         mlog.debug(f'{msg}: `{join_args(args)}` -> {excp}')
         raise
 
-    rc, out, err = p.returncode, o.strip(), e.strip()
     mlog.debug('-----------')
-    mlog.debug(f'{msg}: `{join_args(args)}` -> {rc}')
-    if out:
-        mlog.debug(f'stdout:\n{out}\n-----------')
-    if err:
-        mlog.debug(f'stderr:\n{err}\n-----------')
+    mlog.debug(f'{msg}: `{join_args(args)}` -> {p.returncode}')
+    if o:
+        mlog.debug(f'stdout:\n{o.strip()}\n-----------')
+    if e:
+        mlog.debug(f'stderr:\n{e.strip()}\n-----------')
```

Comment:
I don't think it matters, but this is a slight behavior change: before there would be no debug message for stdout or stderr being only whitespace. Now the `stdout:` and `stderr:` headers are printed anyway.

Reply:
Now that I think about it, I believe that was actually my original reason for writing the original version the way I did. Drat.